core/src/main/java/org/apache/calcite/sql/dialect/MysqlSqlDialect.java

```diff
@@ -179,6 +179,9 @@ public boolean supportsAliasedValues() {
 
       unparseFloor(writer, call);
       break;
+    case ROW:
+      unparseRow(writer, call);
```

Comment:
Thanks for the fix @quxiucheng, i think this patch can be promoted a little, we can move this logic to the parent class `SqlDialect`, you can get the `SqlConformance` from the dialect and decide if we should generates the `ROW` keyword based on method `SqlConformance#allowExplicitRowValueConstructor `.

[1] https://github.com/apache/calcite/blob/4d1c3e54fc4172c7ff00db3326823c42f237cf04/core/src/main/java/org/apache/calcite/sql/validate/SqlConformance.java#L296

Reply:
done.
but  the class `SqlDialect` can't get the method `SqlConformance#allowExplicitRowValueConstructor` correctly. I added a field in the class `SqlDialect`, Same as the field 'SqlConformance#allowExplicitRowValueConstructor'

Reply:
What do you mean by "can not get method correctly", for MySQL, the conformance is `SqlConformanceEnum.MYSQL_5` which i think is correct.

[1] https://github.com/apache/calcite/blob/4d1c3e54fc4172c7ff00db3326823c42f237cf04/core/src/main/java/org/apache/calcite/sql/SqlDialect.java#L1092

Reply:
We have two ways of creating dialects.
1. XxxSqlDialect.DEFAULT
2. SqlDialect.DatabaseProduct.XXX.getDialect()

If we only call `org.apache.calcite.sql.SqlDialect#getConformance` this method to get `conformance`.
Only compatible `SqlDialect.DatabaseProduct.XXX.getDialect()` method. So I reset a parameter `allowExplicitRowValueConstructor` to be compatible with both ways of creating the dialect.

Reply:
`XxxSqlDialect.DEFAULT` also works, if we have set up the `DatabaseProduct` correctly. There is no overridden of method `SqlDialect#getConformance`, so i believe the behavior is correct.

Reply:
I see.Thanks.

I have a question:

the method `SqlConformanceEnum#allowExplicitRowValueConstructor`
```
  public boolean allowExplicitRowValueConstructor() {
    switch (this) {
    case DEFAULT:
    case LENIENT:
      return true;
    default:
      return false;
    }
  }
```
Only `SqlConformanceEnum#DEFAULT` and `SqlConformanceEnum#LENIENT` return values are true.
If this standard doesn't change. Test classes are mostly used by default method `SqlConformanceEnum#PRAGMATIC_99`,
`unparse` will then have a lot of `ROW` keyword changes. Because they don't support the `ROW` keyword.
Should I add `SqlConformanceEnum#PRAGMATIC_99` to a method `SqlConformanceEnum#allowExplicitRowValueConstructor` or change the test class.

Reply:
The return value "true" means we would print the "ROW" keyword, for `SqlConformanceEnum#PRAGMATIC_2003`, it is expected to not allows that.

Reply:
Okay, I'm going to rewrite the code